source/helpers/DataStructure.js

```diff
@@ -0,0 +1,165 @@
+const createAnswerObject = data => ({
+  field: {
+    id: data?.fieldId ?? null,
+    parentId: data?.parentId ?? null,
+    tags: data?.tags ?? [],
+  },
+  value: data?.value ?? null,
+});
+
+/**
+ * Returns form questions as a flat array
+ * @param {obj} form
+ */
+export const getFormQuestions = form => {
+  const formQuestions = [];
+  if (!form || typeof form !== 'object') {
+    return formQuestions;
+  }
+
+  form.steps.forEach(step => {
+    if (step.questions) {
+      step.questions.forEach(question => {
+        formQuestions.push(question);
+      });
+    }
+  });
+
+  return formQuestions;
+};
+
+/**
+ * Convert answers in context to an array that follows Case API data structure
+ * @param {obj} data
+ * @param {obj} formQuestions
+ */
+export const convertAnswersToArray = (data, formQuestions) => {
+  const answers = [];
+
+  if (!data || typeof data !== 'object' || !formQuestions || typeof formQuestions !== 'object') {
+    return answers;
+  }
+
+  Object.entries(data).forEach(answer => {
+    const [fieldId, value] = answer;
+    const { id, type, tags, ...other } = formQuestions.find(element => element.id === fieldId);
+
+    switch (type) {
+      case 'editableList':
+        if (value === undefined) {
+          return;
+        }
```

Comment:
Can you do this check outside before the switch statement? Since you repeat it for all cases except default, where I'm unsure if it also should be...